flang/lib/Optimizer/Transforms/AddDebugInfo.cpp

```diff
@@ -286,10 +286,11 @@ mlir::LLVM::DIModuleAttr AddDebugInfoPass::getOrCreateModuleAttr(
     modAttr = iter->getValue();
   } else {
     modAttr = mlir::LLVM::DIModuleAttr::get(
-        context, fileAttr, scope, mlir::StringAttr::get(context, name),
+        context, decl ? nullptr : fileAttr, decl ? nullptr : scope,
```

Comment:
nit: I don't think it is obvious that setting decl isn't sufficient on its own. Please could you add a comment explaining why `fileAttr` and `line` aren't set when `decl == true`